### PR TITLE
Transformer Ranker & Generator

### PR DESCRIPTION
**Pull request by stephenroller**
Let's make it public!

Odds and ends left before this merges:
- [x] testing (again)
- [x] add a warning this is in beta
- [x] clean up redundant options between the two models
- [x] clean up `-m transformer` call to separate into ranker and generator. Discussion from #1260 

anything else?

### CODE REVIEW

File: parlai/agents/transformer/transformer.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.agents import Agent
+from parlai.core.utils import warn_once
+
+warn_once(
+    "Public release transformer models are currently in beta. The name of "
+    "command line options may disappear before a stable release. We welcome "
```
Comment (stephenroller) [inline parlai/agents/transformer/transformer.py]: edit

File: parlai/agents/transformer/transformer.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    argparser.add_argument('--n-heads', type=int, default=3)
+    argparser.add_argument('--learn-positional-embeddings', type='bool', default=False)
+    argparser.add_argument('--embeddings-scale', type='bool', default=True)
+
+
+class Transformer(Agent):
+    def __init__(self, opt, shared):
+        raise RuntimeError(
+            "`--model transformer` is not a valid choice. Please select either "
+            "`--model transformer/generator` or `--model transformer/generator"
```
Comment (stephenroller) [inline parlai/agents/transformer/transformer.py]: edit

File: parlai/agents/transformer/ranker.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+from parlai.core.torch_ranker_agent import TorchRankerAgent
+from parlai.core.utils import padded_3d
+from .transformer import add_common_cmdline_args
+from .modules import TransformerMemNetModel
+
+
+class RankerAgent(TorchRankerAgent):
```
Comment (stephenroller) [inline parlai/agents/transformer/ranker.py]: move into separate folders I guess.

**Comment by emilydinan** [comment]
awesome!

**Comment by stephenroller** [comment]
no travis?

**Comment by alexholdenmiller** [comment]
it passed: https://travis-ci.org/facebookresearch/ParlAI/builds/455533159

ready to merge?

**Pull request merged**